keps/sig-scheduling/3022-min-domains-in-pod-topology-spread/README.md

```diff
@@ -0,0 +1,502 @@
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+A new field `minDomains` is introduced to `PodSpec.TopologySpreadConstraint[*]` to limit
+the minimum number of topology domains. It functions in a mandatory or best-efforts manner,
+depending on the type of `WhenUnsatisfiable`.
+
+## Motivation
+
+Pod Topology Spread has [`maxSkew` parameter](https://github.com/kubernetes/enhancements/tree/11a976c74e1358efccf251d4c7611d05ce27feb3/keps/sig-scheduling/895-pod-topology-spread#maxskew), which control the degree to which Pods may be unevenly distributed.
+But, there isn't a way to control the number of domains over which we should spread.
+In some cases, users want to force spreading Pods over a minimum number of domains and, if there aren't enough already present, make the cluster-autoscaler provision them.
+
+### Goals
+
+- Users can specify `minDomains` to limit the number of domains.
+- Users can use it as a mandatory requirement with `WhenUnsatisfiable=DoNotSchedule`.
+- Users can use it as a best-efforts manner with `WhenUnsatisfiable=ScheduleAnyway`.
+
+### Non-Goals
+
+- Add new field to limit the maximum number of topology domains.
+
+## Proposal
+
+### User Story
+
+I am using cluster autoscaler and I want to force spreading a deployment over at least 5 Nodes.
+
+## Design Details
+
+Users can define a minimum number of domains with `minDomains` parameter.
+
+Pod Topology Spread has the semantics of "global minimum", which means the minimum number of pods that match the label selector in a topology domain.
+
+When the number of domains that have matching Pods is less than `minDomains`,
+Pod Topology Spread treats "global minimum" as 0.
```

Comment:
```suggestion
Pod Topology Spread treats "global minimum" as 0; otherwise, "global minimum"
is equal to the number of matching pods on a domain.
```

Reply:
the minimum number of matching pods on a domain.